rct229/rulesets/ashrae9012019/section19/section19rule4.py

```diff
@@ -0,0 +1,321 @@
+                        or is_building_area_MF_dormitory_or_hotel_b
+                    )
+
+                def get_manual_check_required_msg(
+                    self, context, calc_vals=None, data=None
+                ):
+                    space_b = context.baseline
+                    space_id_b = space_b["id"]
+
+                    inf_pass_cooling_b = calc_vals["inf_pass_cooling_b"]
+                    occ_pass_cooling_b = calc_vals["occ_pass_cooling_b"]
+                    int_lgt_pass_cooling_b = calc_vals["int_lgt_pass_cooling_b"]
+                    misc_pass_cooling_b = calc_vals["misc_pass_cooling_b"]
+
+                    if all(
+                        [
+                            inf_pass_cooling_b,
+                            occ_pass_cooling_b,
+                            int_lgt_pass_cooling_b,
+                            misc_pass_cooling_b,
+                        ]
+                    ):
+                        be_verb = "was"
+                    else:
+                        be_verb = "was not"
+
+                    return f"It is not clear from the RMD if {space_id_b} is a dwelling unit. If it is a dwelling unit it is required that it be modeled following the rule that for baseline cooling sizing runs in residential dwelling units, the infiltration, occupants, lighting, gas and electricity design day cooling schedules shalled be modeled using an equipment hourly schedule that is the same as the most used hourly weekday schedule from the annual simulation. This rule {be_verb} followed for this space if applicable."
+
+                def rule_check(self, context, calc_vals=None, data=None):
+                    inf_pass_cooling_b = calc_vals["inf_pass_cooling_b"]
+                    occ_pass_cooling_b = calc_vals["occ_pass_cooling_b"]
+                    int_lgt_pass_cooling_b = calc_vals["int_lgt_pass_cooling_b"]
+                    misc_pass_cooling_b = calc_vals["misc_pass_cooling_b"]
+
+                    return all(
+                        [
+                            inf_pass_cooling_b,
+                            occ_pass_cooling_b,
+                            int_lgt_pass_cooling_b,
+                            misc_pass_cooling_b,
+                        ]
+                    )
+
+                def get_fail_msg(self, context, calc_vals=None, data=None):
+                    space_b = context.baseline
+                    space_id_b = space_b["id"]
+
+                    inf_pass_cooling_b = calc_vals["inf_pass_cooling_b"]
+                    occ_pass_cooling_b = calc_vals["occ_pass_cooling_b"]
+                    int_lgt_pass_cooling_b = calc_vals["int_lgt_pass_cooling_b"]
+                    misc_pass_cooling_b = calc_vals["misc_pass_cooling_b"]
+
+                    inf_msg = None if inf_pass_cooling_b else "infiltration"
+                    occ_msg = None if occ_pass_cooling_b else "occupants"
+                    light_msg = None if int_lgt_pass_cooling_b else "lighting"
```

Comment:
Will None being empty string or it prints as "None" in the join function?

Reply:
I just found that ```None``` should've been an empty string b/c ```None``` in the join function caused an error. Addressed.